src/main/java/Command.java

```diff
@@ -0,0 +1,15 @@
+import java.util.Optional;
+
+/**
+ * This allows commands, as indicated by a user's input, to be executed.
+ */
+public interface Command {
+    /**
+     * Executes the intended command.
+     *
+     * @param taskList task list which <code>this</code> task should be added to
+     * @param storage storage which stores all tasks on the local hard disk, if any
+     * @throws Exception if command cannot be executed
+     */
+    public void execute(TaskList taskList, Optional<Storage> storage) throws Exception;
```

Comment:
It would be better if you threw an explicit Exception instead of the catchall.

Reply:
I see! I have thought of that. But do you think that it is too verbose if I explicitly list all exceptions only to perform the same action for each exception later?

Reply:
Yeah it can be annoying to. But I think it’s a good practice to have.